src/hooks/useLifecycleLogger.ts

```diff
@@ -0,0 +1,42 @@
+import type { EffectCallback } from 'react';
+import { useRef, useEffect } from 'react';
+
+const useEffectOnce = (effect: EffectCallback) => {
+  useEffect(effect, []);
+};
+
+export function useFirstMountState(): boolean {
+  const isFirst = useRef(true);
+
+  if (isFirst.current) {
+    isFirst.current = false;
+
+    return true;
+  }
+
+  return isFirst.current;
+}
+
+const useUpdateEffect: typeof useEffect = (effect, deps) => {
+  const isFirstMount = useFirstMountState();
+
+  useEffect(() => {
+    if (!isFirstMount) {
+      return effect();
+    }
+  }, deps);
+};
+
+const useLifecycleLogger = (componentName: string = 'Component', ...rest) => {
+  useEffectOnce(() => {
```

Comment:
We don't need to use `useEffectOnce` I believe. We are already using `useUpdateEffect` so that is fine.